### DISCUSSION TITLE
Builtin File Filter for a custom rector rule

### DISCUSSION THREAD

**Discussion by staabm** [rectorphp/rector#6397] [Q&A]
I am in the process of building a custom rector rule.

I want the rule to skip certain files since the rule itself is only useful in a specific context.
Already tried things like `$this->currentFileProvider->getFile()->getSmartFileInfo()->getRelativeFilePath()` but this gets me a temporary file-path like `..../Local/Temp/_temp_fixture_easy_testing/input_7cfc30fd43dbaab65d33_complex-prop.php`, not the path of the original file.

I don't want the user of my rule needs to specifiy the very specific filesystem related needs for my rule.

is there an easy way to get the path to the file my rector rule is beeing applied on?

**Comment by TomasVotruba** [comment]
This seems duplicate of https://github.com/rectorphp/rector/discussions/6245

In the `*Rector.php` rule, the `File` is always present, via `$this->file->`

Test fixture generate would need an upgrade for this change.
